sdk/storage/azure_storage_blob/src/clients/page_blob_client.rs

```diff
@@ -0,0 +1,158 @@
+        NoFormat, RequestContent, Response, Url,
+    },
+    Bytes, Result,
+};
+use std::sync::Arc;
+
+/// A client to interact with a specific Azure storage Page blob, although that blob may not yet exist.
+pub struct PageBlobClient {
+    pub(crate) endpoint: Url,
+    pub(crate) client: GeneratedPageBlobClient,
+}
+
+impl PageBlobClient {
+    /// Creates a new PageBlobClient, using Entra ID authentication.
+    ///
+    /// # Arguments
+    ///
+    /// * `endpoint` - The full URL of the Azure storage account, for example `https://myaccount.blob.core.windows.net/`
+    /// * `container_name` - The name of the container containing this Page blob.
+    /// * `blob_name` - The name of the Page blob to interact with.
+    /// * `credential` - An implementation of [`TokenCredential`] that can provide an Entra ID token to use when authenticating.
+    /// * `options` - Optional configuration for the client.
+    pub fn new(
+        endpoint: &str,
+        container_name: String,
+        blob_name: String,
+        credential: Arc<dyn TokenCredential>,
+        options: Option<PageBlobClientOptions>,
+    ) -> Result<Self> {
+        let mut options = options.unwrap_or_default();
+
+        let storage_headers_policy = Arc::new(StorageHeadersPolicy);
+        options
+            .client_options
+            .per_call_policies
+            .push(storage_headers_policy);
+
+        let oauth_token_policy = BearerTokenCredentialPolicy::new(
+            credential.clone(),
+            ["https://storage.azure.com/.default"],
+        );
+        options
+            .client_options
+            .per_try_policies
+            .push(Arc::new(oauth_token_policy) as Arc<dyn Policy>);
+
+        let client = GeneratedPageBlobClient::new(
+            endpoint,
+            credential.clone(),
+            container_name.clone(),
+            blob_name.clone(),
+            Some(options),
+        )?;
+        Ok(Self {
+            endpoint: endpoint.parse()?,
+            client,
+        })
+    }
+
+    /// Gets the endpoint of the Storage account this client is connected to.
+    pub fn endpoint(&self) -> &Url {
+        &self.endpoint
+    }
+
+    /// Gets the container name of the Storage account this client is connected to.
+    pub fn container_name(&self) -> &str {
+        &self.client.container_name
+    }
+
+    /// Gets the blob name of the Storage account this client is connected to.
+    pub fn blob_name(&self) -> &str {
+        &self.client.blob_name
+    }
+
+    /// Creates a new Page blob.
+    ///
+    /// # Arguments
+    ///
+    /// * `content_length` - The maximum size for the Page blob, up to 1TB. The page blob size must
+    ///   be aligned to a 512-byte boundary.
+    /// * `options` - Optional parameters for the request. See [`azure_storage_blob::storage_traits::PageBlobClientCreateOptionsExt`](crate::storage_traits::PageBlobClientCreateOptionsExt) for additional usage helpers.
```

Comment:
This is how we plan to ensure discoverability. Docs render:
![image](https://github.com/user-attachments/assets/43bca67a-1a4b-4286-8fb5-f5a40c862276)

Reply:
Don't make the whole path a link. For one, if it's already in scope, you don't need to qualify it. If not, use markdown to shorten it:
```suggestion
    /// * `options` - Optional parameters for the request. See [`PageBlobClientCreateOptionsExt`](crate::storage_traits::PageBlobClientCreateOptionsExt) for additional usage helpers.
```